### PR TITLE
[infra] Add armv7em platform

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This PR adds cortex-m7 platform.

issue #9468 

ONE-DCO-1.0-Signed-off-by: Artem Balyshev a.balyshev@partner.samsung.com

### CODE REVIEW

**Review by binarman — DISMISSED** [review]
LGTM

**Comment by hseok-oh** [comment]
What is purpose of this PR? If you want to set cmake option for specific CPU, you can pass explicitly on cmake configuration. My concern is `cortex-m7` is not platform name (ex. get by `uname` command).

**Comment by chunseoklee** [comment]
@BalyshevArtem As @hseok-oh commented, Could you please proceed with "armv7-m" ?

**Comment by BalyshevArtem** [comment]
@hseok-oh, @chunseoklee, could you please help, I actually don't know good this part, what should I do to make build for `cortex-m7`, because now without  these changes it doesn't build, it wants changes in `infra/cmake/modules/IdentifyPlatform.cmake` and in `infra/nncc/cmake/options/options_cortex-m7-generic.cmake` ?
thank you in advance :)

**Comment by chunseoklee** [comment]
@BalyshevArtem Could you share your local build configuration (and target information) ? 

> infra/nncc/cmake/options/options_cortex-m7-generic.cmake

It could be options_armv4m-linux , options_armv4m-tizenrt. Is there any issue if we use those target_os name ?

**Comment by seanshpark** [comment]
> It could be options_armv4m-linux , options_armv4m-tizenrt. Is there any issue if we use those target_os name ?

I'm not good at this but Cortex-M seems to be ARMv7E-M
- https://en.wikipedia.org/wiki/ARM_Cortex-M
   - _the Cortex-M4 / Cortex-M7 implements the ARMv7E-M architecture,_
- IMHO, we may use `options_armv7em-`

**Comment by hseok-oh** [comment]
>@hseok-oh, @chunseoklee, could you please help, I actually don't know good this part, what should I do to make build for cortex-m7, because now without these changes it doesn't build, it wants changes in infra/cmake/modules/IdentifyPlatform.cmake and in infra/nncc/cmake/options/options_cortex-m7-generic.cmake ?

If you want to pass different compile option for `cortex-m7`, we may need to introduce new config file, and following new platform name. If you want to set different cmake option, you can set by cmake configuration option, such as

```
./nncc configure -D{OPTION1}=ON -D{OPTION2}=OFF ...
```

File: infra/cmake/modules/IdentifyPlatform.cmake [hunk 1]
```diff
@@ -49,6 +51,8 @@ endif()
 
 if("${TARGET_ARCH}" STREQUAL "x86_64")
   set(TARGET_ARCH_BASE ${TARGET_ARCH})
+elseif("${TARGET_ARCH}" STREQUAL "armv7em")
+  set(TARGET_ARCH_BASE "arm")
```
Comment (BalyshevArtem) [inline infra/cmake/modules/IdentifyPlatform.cmake:55]: @hseok-oh, I introduced armv7em arch, according to https://github.com/Samsung/ONE/pull/9477#issuecomment-1193400713 comment

**Review by hseok-oh — APPROVED** [review]

**Comment by chunseoklee** [comment]
@BalyshevArtem Could you please share any overall draft(including additional conf files) if possible ?

**Comment by chunseoklee** [comment]
PTAL @binarman

**Comment by BalyshevArtem** [comment]
@chunseoklee, you can find draft here - https://github.com/BalyshevArtem/ONE/tree/draft_luci_micro_building.
 the only change from this pr is https://github.com/BalyshevArtem/ONE/commit/882e81bec39fbe1b9c20c4658976cec5f0065803: I added `-DTARGET_ARCH=armv7em` line for correct building of luci-micro

**Review by binarman — APPROVED** [review]

File: infra/cmake/modules/IdentifyPlatform.cmake [hunk 2]
```diff
@@ -35,6 +35,8 @@ endif()
 
 if("${HOST_ARCH}" STREQUAL "x86_64")
   set(HOST_ARCH_BASE ${HOST_ARCH})
+elseif("${HOST_ARCH}" STREQUAL "armv7em")
+  set(HOST_ARCH_BASE "arm")
```
Comment (binarman) [inline infra/cmake/modules/IdentifyPlatform.cmake:39]: small note: this is an interesting case to use embedded MCU for build :)

**Pull request merged**